src/transform/classify.rs

```diff
@@ -34,56 +65,194 @@ fn classify_one(mut event: Event, classes: &Vec<(String, Regex)>) -> Event {
     // An event can only have one category, although the category may have a hierarchy,
     // for instance: "Work -> ActivityWatch -> aw-server-rust"
     // A category is chosed out of the tags used some rule (such as picking the one that's deepest in the hierarchy)
-    let category = choose_category(&tags);
-    event.data.insert("$category".into(), serde_json::json!(category));
+    let category = _choose_category(&tags);
+    event
+        .data
+        .insert("$category".into(), serde_json::json!(category));
     event
 }
 
-fn choose_category(tags: &HashSet<String>) -> String {
-    tags.iter().fold(&"Uncategorized".to_string(), |acc: &String, item: &String| {
-        if item.starts_with('#') {
-            // If tag is not a category, then skip.
-            acc
-        } else if item.matches("->").count() >= acc.matches("->").count() {
-            // If tag is category with greater or equal depth than current, then choose the new one instead.
-            item
-        } else {
-            acc
+/// Categorizes a list of events
+///
+/// An event can only have one category, although the category may have a hierarchy,
+/// for instance: "Work -> ActivityWatch -> aw-server-rust"
+/// A category is chosed out of the tags used some rule (such as picking the one that's deepest in the hierarchy)
+// TODO: Classes should be &Vec<(String, Rule)>
+pub fn categorize(mut events: Vec<Event>, rules: &Vec<(Vec<String>, Regex)>) -> Vec<Event> {
+    let mut classified_events = Vec::new();
+    for event in events.drain(..) {
+        classified_events.push(categorize_one(event, rules));
+    }
+    return classified_events;
+}
+
+// TODO: Classes should be &Vec<(String, Rule)>
+pub fn autotag(mut events: Vec<Event>, rules: &Vec<(String, Regex)>) -> Vec<Event> {
```

Comment:
autotag makes it sound like it comes up with the tags itself (which would be really cool, but unfortunately that's not what it does).
Just "match_tags" would better fit as a name.

Reply:
I renamed it to just `tag` because otherwise, I feel like we should also rename the `categorize` function to `match_category` and that just seems unnecessarily wordy.